src/state_machine.rs

```diff
@@ -246,7 +242,7 @@ impl StateMachine {
     // Construct a new StateMachine by passing a function returning the initial state.
     pub fn new<F>(
         init_state: F,
-        pub_id: PublicId,
+        sec_id: SecretKeys,
```

Comment:
Not checking this thoroughly just now as other fundamental changes are required as put in the review comments elsewhere, but do pls go over this and see this is what we are saying we need - a prior pub stuff getting replace by secret stuff is suspicious code. Might be totally legit but pls see what and why are we wanting this.

Reply:
In this case it makes sense cause `crust::Service` requires `SecretKeys` to do data encryption. Before Crust didn't encrypt anything.